### PR TITLE
XW-2226 | Add new page experiment

### PR DESCRIPTION
**Pull request by kvas-damian**
Show add new page button in different variations

https://wikia-inc.atlassian.net/browse/XW-2226

Ping @Wikia/x-wing

### CODE REVIEW

File: skins/oasis/js/Tracking.js [hunk 1]
```diff
@@ -259,6 +259,16 @@ jQuery(function ($) {
 		}
 	});
 
+	// TODO remove after XW-2226 is done
```
Comment (WikiaTech) [inline skins/oasis/js/Tracking.js:262]: INFO Complete the task associated to this TODO comment. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS1135)

File: skins/oasis/js/Tracking.js [hunk 2]
```diff
@@ -259,6 +259,16 @@ jQuery(function ($) {
...
+	$('.add-new-page-experiment-element.createpage').on('mousedown', function (event) {
+		track({
+			browserEvent: event,
+			category: 'contribute',
+			label: 'add-a-page'
+		});
+	});
+	// TODO remove end
```
Comment (WikiaTech) [inline skins/oasis/js/Tracking.js:270]: INFO Complete the task associated to this TODO comment. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS1135)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 10 issues
* MAJOR 7 major
* MINOR 1 minor
* INFO 2 info

Watch the comments in this conversation to review them.

#### 8 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [Tracking.js#L73](https://github.com/Wikia/app/blob/e0cf79dd301ee0eddd40d55d1aa90803f96907a1/skins/oasis/js/Tracking.js#L73): Function has a complexity of 15 which is greater than 10 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AFunctionComplexity)
1. MAJOR [Tracking.js#L96](https://github.com/Wikia/app/blob/e0cf79dd301ee0eddd40d55d1aa90803f96907a1/skins/oasis/js/Tracking.js#L96): Add a "default" clause to this "switch" statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
1. MAJOR [Tracking.js#L232](https://github.com/Wikia/app/blob/e0cf79dd301ee0eddd40d55d1aa90803f96907a1/skins/oasis/js/Tracking.js#L232): Add a "default" clause to this "switch" statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
1. MAJOR [Tracking.js#L596](https://github.com/Wikia/app/blob/e0cf79dd301ee0eddd40d55d1aa90803f96907a1/skins/oasis/js/Tracking.js#L596): Add a "default" clause to this "switch" statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
1. MAJOR [Tracking.js#L616](https://github.com/Wikia/app/blob/e0cf79dd301ee0eddd40d55d1aa90803f96907a1/skins/oasis/js/Tracking.js#L616): Function has a complexity of 12 which is greater than 10 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AFunctionComplexity)
1. MAJOR [Tracking.js#L658](https://github.com/Wikia/app/blob/e0cf79dd301ee0eddd40d55d1aa90803f96907a1/skins/oasis/js/Tracking.js#L658): Function has a complexity of 16 which is greater than 10 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AFunctionComplexity)
1. MAJOR [Tracking.js#L673](https://github.com/Wikia/app/blob/e0cf79dd301ee0eddd40d55d1aa90803f96907a1/skins/oasis/js/Tracking.js#L673): Add a "default" clause to this "switch" statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
1. MINOR [Tracking.js#L596](https://github.com/Wikia/app/blob/e0cf79dd301ee0eddd40d55d1aa90803f96907a1/skins/oasis/js/Tracking.js#L596): Replace this "switch" statement with "if" statements to increase readability. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS1301)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 16 issues
- CRITICAL 2 critical
- MAJOR 10 major
- MINOR 2 minor
- INFO 2 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. CRITICAL [CreatePage.js#L346](https://github.com/Wikia/app/blob/2fa24b55d507ee8b1c90dac1b4c87ef49ab378ec/extensions/wikia/CreatePage/js/CreatePage.js#L346): Remove this "===" check; it will always be false. Did you mean to use "=="? [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS3403)
2. CRITICAL [CreatePage.js#L346](https://github.com/Wikia/app/blob/2fa24b55d507ee8b1c90dac1b4c87ef49ab378ec/extensions/wikia/CreatePage/js/CreatePage.js#L346): Change this condition so that it does not always evaluate to "false". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS2583)
3. MAJOR [CreatePage.js#L43](https://github.com/Wikia/app/blob/2fa24b55d507ee8b1c90dac1b4c87ef49ab378ec/extensions/wikia/CreatePage/js/CreatePage.js#L43): Function has a complexity of 14 which is greater than 10 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AFunctionComplexity)
4. MAJOR [CreatePage.js#L135](https://github.com/Wikia/app/blob/2fa24b55d507ee8b1c90dac1b4c87ef49ab378ec/extensions/wikia/CreatePage/js/CreatePage.js#L135): Remove the unused function parameter "event". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
5. MAJOR [CreatePage.js#L205](https://github.com/Wikia/app/blob/2fa24b55d507ee8b1c90dac1b4c87ef49ab378ec/extensions/wikia/CreatePage/js/CreatePage.js#L205): Restrict what this loop acts on by testing each property. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AForIn)
6. MAJOR [Tracking.js#L73](https://github.com/Wikia/app/blob/2fa24b55d507ee8b1c90dac1b4c87ef49ab378ec/skins/oasis/js/Tracking.js#L73): Function has a complexity of 15 which is greater than 10 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AFunctionComplexity)
7. MAJOR [Tracking.js#L96](https://github.com/Wikia/app/blob/2fa24b55d507ee8b1c90dac1b4c87ef49ab378ec/skins/oasis/js/Tracking.js#L96): Add a "default" clause to this "switch" statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
8. MAJOR [Tracking.js#L232](https://github.com/Wikia/app/blob/2fa24b55d507ee8b1c90dac1b4c87ef49ab378ec/skins/oasis/js/Tracking.js#L232): Add a "default" clause to this "switch" statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
9. MAJOR [Tracking.js#L596](https://github.com/Wikia/app/blob/2fa24b55d507ee8b1c90dac1b4c87ef49ab378ec/skins/oasis/js/Tracking.js#L596): Add a "default" clause to this "switch" statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
10. MAJOR [Tracking.js#L616](https://github.com/Wikia/app/blob/2fa24b55d507ee8b1c90dac1b4c87ef49ab378ec/skins/oasis/js/Tracking.js#L616): Function has a complexity of 12 which is greater than 10 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AFunctionComplexity)

File: extensions/wikia/CreatePage/js/CreatePage.js [hunk 3]
```diff
@@ -295,7 +295,7 @@ var CreatePage = {
 		'use strict';
 		CreatePage.context = context;
 
-		$( '.WikiaMenuElement .createpage' ).click(function() {
+		$( '.createpage' ).click(function() {
```
Comment (rogatty) [inline extensions/wikia/CreatePage/js/CreatePage.js:298]: What do you think about adding an id and using it here?

File: skins/oasis/modules/templates/ContributeMenu_Index.php [hunk 4]
```diff
@@ -1,5 +1,16 @@
 <?php
 
+// TODO remove after XW-2226 is done
+echo F::app()->renderView('MenuButton', 'Index', array(
```
Comment (hakubo) [inline skins/oasis/modules/templates/ContributeMenu_Index.php:4]: [] ! :)

**Review by hakubo — APPROVED** [review]

**Review by rogatty — APPROVED** [review]

File: extensions/wikia/CreatePage/js/CreatePage.js (hunk 3 above)
Comment (kvas-damian) [reply extensions/wikia/CreatePage/js/CreatePage.js:298]: id isn't probably the best idea when we have multiple "Create a Page" buttons - not only for ABTest

File: skins/oasis/modules/templates/ContributeMenu_Index.php (hunk 4 above)
Comment (kvas-damian) [reply skins/oasis/modules/templates/ContributeMenu_Index.php:4]: We'll remove it after ABTest ;)

**Pull request merged**